cloud2cloud-connector/service/deviceSubscription.go

```diff
@@ -40,9 +41,9 @@ func (s *SubscriptionManager) SubscribeToDevice(ctx context.Context, deviceID st
 		linkedCloud:   linkedCloud,
 		subscription:  sub,
 	}
-	err = s.cache.Add(correlationID, data, cache.DefaultExpiration)
-	if err != nil {
-		return fmt.Errorf("cannot cache subscription for device subscriptions: %w", err)
+	_, loaded := s.cache.LoadOrStore(correlationID, cache.NewElement(data, time.Now().Add(time.Minute*10), nil))
```

Comment:
Maybe some constant for `time.Minute*10`, the value is used in multiple places when inserting into the cache.

Reply:
done